odoo/addons/base/models/ir_model.py

```diff
@@ -1368,6 +1368,29 @@ def unlink(self):
 
     def _process_ondelete(self):
         """ Process the 'ondelete' of the given selection values. """
+        def write(records, col, value):
+            if not records:
+                return
+            try:
+                with self.env.cr.savepoint():
+                    records.write({col: value})
+            except Exception as e:
+                # going through the ORM failed, probably because of a write override that raises
+                # an error for this specific field or possibly a constraint.
+                _logger.warning(
+                    "Could not fulfill ondelete action for field %s of model %s, "
+                    "attempting ORM bypass...", col, records._name,
+                )
+                query = sql.SQL("UPDATE {} SET {}=%s WHERE id IN %s").format(
+                    sql.Identifier(records._table),
+                    sql.Identifier(col),
+                )
+                # if this fails then we're shit out of luck and there's nothing we can do
+                # except fix on a case-by-case basis
+                with self.env.cr.savepoint():
```

Comment:
this savepoint is not useful here, as the caller does not expect the transaction to have rolled back